dlopen/dlopen.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO(iaguis) move to a better place
```

Comment:
(Maybe not so) wild idea for the future - write a program that takes a file like:
```
#include <this>
#include <that>
int
c_function1_to_wrap(char* param1, int param2);
void
c_function2_to_wrap(void);
```
and would output a go file like:
```go
package something

// #cgo LDFLAGS: -ldl
// #include <this>
// #include <that>
// int
// my_c_function1_to_wrap(void* f, char* param1, int param2)
// {
//     int (*rf)(char*, int);
//
//     rf = f;
//     return rf(param1, param2);
// }
//
// void
// my_c_function2_to_wrap(void)
// {
//     void (*rf)(void);
//
//     rf = f;
//     rf();
// }
import "C"
```
And then in other file in the same package:
```go
//go:generate <that_tool> <that_file> -o generated_c_boilerplate.go // +maybe other flags
…
```
Maybe that would partially save us from writing lots of boilerplate code.